### PR TITLE
588 docstrings

### PR DESCRIPTION
**Pull request by DaJansenGit**

### CODE REVIEW

File: bim2sim/tasks/bps/sb_correction.py [hunk 1]
```diff
@@ -41,14 +41,40 @@ class CorrectSpaceBoundaries(ITask):
...
+        Elements that are relevant for exporting EnergyPlus Input Files within the
+        Plugin EnergyPlus. This geometric preprocessing mainly relies on shape
+        manipulations with OpenCascade (OCC). This task is the first in the
+        BIM2SIM PluginEnergyPlus and must be executed before exporting the
+        EnergyPlus Input file (IDF).
+        This task starts with linking the space boundaries to the dictionary
+        of elements. Additionally, geometric preprocessing operations are
+        executed, like moving opening elements to their parent surfaces (
+        unless they are already coplanar), the surface orientation of space
+        boundaries are fixed, and non-convex boundaries are fixed.
```
Comment (veronikarichter) [inline bim2sim/tasks/bps/sb_correction.py:65]:
```suggestion
        """Geometric preprocessing for BPS.

        This module contains all functions for geometric preprocessing of the BIM2SIM
        Elements that are relevant for exporting BPS Input Files within the
        Plugins EnergyPlus, Comfort and TEASER. This geometric preprocessing mainly 
        relies on shape manipulations with OpenCascade (OCC). 
        This task starts with linking the space boundaries to the dictionary
        of elements. Additionally, geometric preprocessing operations are
        executed, like moving opening elements to their parent surfaces (
        unless they are already coplanar), the surface orientation of space
        boundaries are fixed, and non-convex boundaries are fixed.
```

File: bim2sim/tasks/hvac/dead_ends.py [hunk 2]
```diff
@@ -54,17 +74,27 @@ def identify_dead_ends(graph: HvacGraph) -> list:
     def decide_dead_ends(graph: HvacGraph, pot_dead_ends: list,
                          playground: Playground = None,
                          force: bool = False) -> [{HvacGraph}, int]:
-        """Decides for all dead ends whether they are consumers or dead ends.
+        """Decides for all dead ends whether they are consumers or dead ends in
```
Comment (veronikarichter) [inline bim2sim/tasks/hvac/dead_ends.py:77]: single line

File: bim2sim/tasks/hvac/expansiontanks.py [hunk 3]
```diff
@@ -48,18 +65,29 @@ def decide_expansion_tanks(
             potential_expansion_tanks: set,
             playground: Playground = None,
             force: bool = False) -> [HvacGraph, int]:
-        """Delete the found expansions tanks. If force is false a decision will
-         be called
+        """Decide and handle the removal of potential expansion tanks in the
```
Comment (veronikarichter) [inline bim2sim/tasks/hvac/expansiontanks.py:68]: as above

File: bim2sim/tasks/hvac/reduce.py [hunk 4]
```diff
@@ -9,12 +9,27 @@
 
 
 class Reduce(ITask):
-    """Reduce number of elements by aggregation."""
 
     reads = ('graph',)
     touches = ('graph',)
 
     def run(self, graph: HvacGraph) -> (HvacGraph,):
+        """Apply aggregations to reduce the number of elements in the HVAC
```
Comment (veronikarichter) [inline bim2sim/tasks/hvac/reduce.py:17]: single line

File: bim2sim/utilities/check_docstrings.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    output = result.stdout
+
+    output_list = output.splitlines()
+
+    functions_without_docstrings = 0
+    last_function_index = 0
+    structured_messages = {}
+    pattern = r'D\d{3}:'
+    for i, line in enumerate(output_list):
+        # | ['C:\\Users\\Arbeit David\\Documents\\02_Git\\bim2sim\\bim2sim\\plugins\\AixLib\\AixLib\\Resources\\Data\\BoundaryConditions\\Validation\\BESTEST\\generateResults.py:1 at module level:'] |
```
Comment (veronikarichter) [inline bim2sim/utilities/check_docstrings.py:18]:
```suggestion

```

File: bim2sim/tasks/bps/disaggr_building_elements.py [hunk 6]
```diff
@@ -0,0 +1,250 @@
+import inspect
```
[the remark this answers is not in the document]
Comment (DaJansenGit) [reply bim2sim/tasks/bps/disaggr_building_elements.py]: you are correct, I deleted it.

File: bim2sim/tasks/hvac/dead_ends.py [hunk 7]
```diff
@@ -5,12 +5,28 @@
 
 
 class DeadEnds(ITask):
-    """Analyses graph network for dead ends and removes ports due to dead ends."""
 
     reads = ('graph',)
     touches = ('graph',)
 
     def run(self, graph: HvacGraph) -> HvacGraph:
+        """Analyses graph network for dead ends and removes ports due to dead
```
[the remark this answers is not in the document]
Comment (DaJansenGit) [reply bim2sim/tasks/hvac/dead_ends.py]: done

File: bim2sim/tasks/hvac/expansiontanks.py (hunk 3 above)
Comment (DaJansenGit) [reply bim2sim/tasks/hvac/expansiontanks.py]: done

File: bim2sim/tasks/hvac/reduce.py (hunk 4 above)
Comment (DaJansenGit) [reply bim2sim/tasks/hvac/reduce.py]: done

File: bim2sim/tasks/bps/sb_correction.py [hunk 8]
```diff
@@ -41,14 +41,38 @@ class CorrectSpaceBoundaries(ITask):
...
+        Args:
+            elements (dict): dictionary in the format dict[guid: element],
+                Dictionary of elements generated in previous IFC-based setup and
+                enrichment tasks. In this task, the elements are enriched with
+                the geometric preprocessed space_boundary items.
+            space_boundaries (dict): dictionary in the format dict[guid:
+                SpaceBoundary], dictionary of IFC-based space boundary elements.
+        """
+
+        logger.info("Geometric preprocessing for EnergyPlus Export started"
```
Comment (veronikarichter) [inline bim2sim/tasks/bps/sb_correction.py:74]:
```suggestion
        "
```

obsolete, see L78

**Review by veronikarichter — CHANGES_REQUESTED** [review]

File: bim2sim/tasks/bps/sb_correction.py (hunk 8 above)
Comment (DaJansenGit) [reply bim2sim/tasks/bps/sb_correction.py]: solved

**Review by veronikarichter — APPROVED** [review]

**Pull request merged**